Add Kubecost construct example

elenalape opened the pull request:
*Description of changes:*

Add a [Kubecost AddOn](https://github.com/kubecost/kubecost-ssp-addon) usage example to `lib/`. 


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

shapirov103 reviewed (commented):
Looks good, everything worked!

A couple of things:

1. I noticed the chart will install prometheus node exporter. Is there a way to supply an option to not install it and leverage an already installed (if one is available)?
2. Can you share links/resources on how to leverage the installed software? Or add an md file under /docs with the references to the github repo and documentation site that explains how to use it? 
3. There are cases when customers cannot leverage an external helm chart repository and instead need to supply their private repo. Do you support this model (from the business perspective)? If yes, do you mind providing customers options to specify the chart repo, e.g similar to this example: https://github.com/shapirov103/ssp-eks-extension/blob/main/lib/index.ts#L22 (deriving from Helm Addon). It will also enable customers to use GitOps to maintain the addon.

shapirov103 commented:
The checks are failing: your branch requires rebasing (merge from main) to address.

elenalape commented:
@shapirov103 hi hi, thanks for the review!

1. [Yes](https://github.com/kubecost/docs/blob/main/getting-started.md#using-an-existing-node-exporter), customers can use their own NE provided it's running on the default port and its endpoint is visible from the namespace where Kubecost is installed. Customers are also welcome use their own Prometheus (including AMP), though it's only officially supported through paid plans. We've added [both of these functionalities](https://github.com/kubecost/kubecost-ssp-addon/blob/main/src/index.ts#L43) to the AddOn and [documented them](https://github.com/kubecost/kubecost-ssp-addon/#customprometheus-string-optional): 

2. The getting started parts in the [product and technical docs](https://docs.kubecost.com/getting-started) contain more details on how to use Kubecost. Added to the [AddOn README](https://github.com/kubecost/kubecost-ssp-addon/#kubecost-addon-for-aws-ssp-cdk-platform).

3. We do support it; updated the [AddOn](https://github.com/kubecost/kubecost-ssp-addon/blob/main/src/index.ts#L24).

shapirov103 commented on lib/kubecost-construct/index.ts:
```diff
@@ -0,0 +1,24 @@
+import * as cdk from '@aws-cdk/core';
+import * as ssp from '@aws-quickstart/ssp-amazon-eks'
+
+import { KubecostAddOn, KubecostAddOnProps } from '@kubecost/kubecost-ssp-addon';
```
Unsure why lint did not catch it (will fix), the KubecostAddOnProps is unused, and github action complains about it.

elenalape replied:
I think this may have been me 😅

shapirov103 commented on lib/kubecost-construct/index.ts:
```diff
@@ -0,0 +1,24 @@
+
+import { KubecostAddOn, KubecostAddOnProps } from '@kubecost/kubecost-ssp-addon';
+
+export default class KubecostConstruct extends cdk.Construct {
```
To reduce verbosity we have introduced the build pattern and would like it to use consistently for all blueprints. Do you mind changing the body of this pattern to this:

```typescript
import * as cdk from '@aws-cdk/core';
import { EksBlueprint } from '@aws-quickstart/ssp-amazon-eks';
import { KubecostAddOn } from '@kubecost/kubecost-ssp-addon';


export default class KubecostConstruct extends cdk.Construct {
    constructor(scope: cdk.Construct, id: string) {
        super(scope, id);
        // AddOns for the cluster
        const stackId = `${id}-blueprint`;

        EksBlueprint.builder()
            .account(process.env.CDK_DEFAULT_ACCOUNT!)
            .region(process.env.CDK_DEFAULT_REGION)
            .addOns(new KubecostAddOn())
            .build(scope, stackId);
    }
}
```

elenalape replied:
sounds good, done!

shapirov103 reviewed (commented):
Looks great and about to merge. One comment (tactical) on the pattern, will merge after.

shapirov103 commented:
> 
> 2. The getting started parts in the [product and technical docs](https://docs.kubecost.com/getting-started) contain more details on how to use Kubecost. Added to the [AddOn README](https://github.com/kubecost/kubecost-ssp-addon/#kubecost-addon-for-aws-ssp-cdk-platform).

Let's get a snippet for the https://github.com/aws-quickstart/ssp-amazon-eks/tree/main/docs/addons page for Kubecost and add these references there as well. 

As a follow up item (roadmap), we should consider a pattern with full ingress enabled leveraging other add-ons that will be turnkey for the customer. e.g. if there is one that 50% of the customer use (or most commonly used configuration).

elenalape commented:
Thanks for the feedback! Changes made and PR submitted https://github.com/aws-quickstart/ssp-amazon-eks/pull/256

shapirov103 reviewed (approved):
LGTM
